- fuente: https://www.kaggle.com/datasets/abdocan/laptop-prices

 # 1 Actividad I- Elecci´on potenciales datasets

 ## Consigna
 1. Identificar 3 datasets que cumplan con las siguientes condiciones: 
 a) almenos 2000 filas
 b) al menos 15 columnas. 
 Pueden buscar en las sigu
ientes fuentes: 
GitLab, Github, Kaggle, Google Dataset Search (Si desean
 trabajar con un archivo propio se puede tambi´ en). 
 Algunas API recomen
dadas para obtener informaci´on: Marvel,PokeApi,CovidTracking,Nomics
 (Criptomonedas),Wheater API

 
 2. Cargar los archivos correspondientes por medio de la librer´ ıa pandas
 3. Describir las variables potencialmente interesantes en cada archivo te
niendo en cuenta el contexto comercial y anal´ ıtico involucrado

# Variables potencialmente interesantes
## Contexto comercial

Las siguientes variables son las que mas afectan el precio o se ven con frecuencia en las laptops con mas valor
 - Ram
 - SSD
 - Cpu brand
 - TypeName

Las siguientes variables se ven con mas frecuencia atribuidas a las laptops de mas bajo costo

In [ ]:
#Librerias 
import pandas as pd

# Carga de datos

In [ ]:
#Lectura de datos
df = pd.read_csv("laptop-price.csv")

#Conversion de Indias rupias a dolares
tasa_cambio = 0.0118025
df["Price"] = df["Price"] * tasa_cambio
df.to_csv("Price", index=False)

#Columnas irrelevantes
df.drop(columns=["Hybrid", "index"], inplace = True)

df

Company            TypeName  Ram  Weight        Price  Touchscreen  Ips  \
0      Apple           Ultrabook    8    1.37   842.446908            0    1   
1      Apple           Ultrabook    8    1.34   565.286913            0    0   
2         HP            Notebook    8    1.86   361.581390            0    0   
3      Apple           Ultrabook   16    1.83  1595.642953            0    1   
4      Apple           Ultrabook    8    1.37  1134.170774            0    1   
...      ...                 ...  ...     ...          ...          ...  ...   
1263  Lenovo  2 in 1 Convertible    4    1.80   401.198134            1    1   
1264  Lenovo  2 in 1 Convertible   16    1.30   942.626963            1    1   
1265  Lenovo            Notebook    2    1.50   144.003719            0    0   
1266      HP            Notebook    6    2.19   480.431621            0    0   
1267    Asus            Notebook    4    2.20   232.040927            0    0   

             ppi              Cpu brand   HDD  SSD  Flash_Storage Gpu brand  \
0     226.983005          Intel Core i5     0  128              0     Intel   
1     127.677940          Intel Core i5     0    0            128     Intel   
2     141.211998          Intel Core i5     0  256              0     Intel   
3     220.534624          Intel Core i7     0  512              0       AMD   
4     226.983005          Intel Core i5     0  256              0     Intel   
...          ...                    ...   ...  ...            ...       ...   
1263  157.350512          Intel Core i7     0  128              0     Intel   
1264  276.053530          Intel Core i7     0  512              0     Intel   
1265  111.935204  Other Intel Processor     0    0             64     Intel   
1266  100.454670          Intel Core i7  1000    0              0       AMD   
1267  100.454670  Other Intel Processor   500    0              0     Intel   

                      os  
0                    Mac  
1                    Mac  
2     Others/No OS/Linux  
3                    Mac  
4                    Mac  
...                  ...  
1263             Windows  
1264             Windows  
1265             Windows  
1266             Windows  
1267             Windows  

[1268 rows x 14 columns]

# Informacion de DataSet

El dataset tiene 1268 filas y 16 columnas

In [12]:
#Dimensiones del DataSet
df.shape

(1268, 16)

# Descripcion de las variables
**Index**: El el indice que realciona la posicion del respectivo dato de una fila.

**Company**: Es la compañia de la laptop

**TypeName**: Es tipo de lapptop

**Ram**: El monsto de ram dado en (GB)

**Weight**: El peso de la laptop en (Kg)

**Price**: El precio de la laptop en USD (inicialmente en IND Rupee)

**TouchScreen**: Indica si la laptop cuanta con tactil en la pantalla (Se muestra para el analisis en binario SI/NO)


**Ips**: Indica si la tablep cuenta con en su pantalla con la tecnologia de "Crtistales liquidos alineados en paralelo". Siglas en ingles (In-Plane Switching). Es un tipo de tecnologia de visualizacion que proporciona una mejor experiencia visual al usuario (Se muestra para el analisis en binario SI/NO)

**ppi**: Indica los pixeles por pulgada con los que cuenta la pantalla de la laptop, entre mas tenga se tendra una mayor calidad de visualizacion de imagen en el monitor.

**Cpu brand:** Es la marca de la unidad de procesamiento. La cpu es la unidad de procesmamiento la cual le permite a la laptop realizar calculos mas complejos y procesar mejor los datos

**HDD**: Unidad de disco duro, es la parte que se encarga del almacenamiento, a diferencia de una SSD, esta cuenta con partes mecanicas, escribe los datos y los almacena magenticamente en discos circulares

**SSD**: Disco de estado solido. Es la unidad de almacenamiento. Cuenta con una mejor tecnologia respecto a la HDD, se caracteriz en su ausencia de partes moviles

**Hybrid**: Sistema de alamcenamiento que combina la unidad de disco duro (HDD) y el disco de estado solido (SSD) (Se muestra para el analisis en binario SI/NO)

**Flash_Storage**: Es una tecnologia de almacenamiento. Hoy en dia no se diferencia del disco de estado solido (SSD), Estado dado en (GB)

**Gpu Brand**: Es es la marca de la unidad de procesamiento de graficos 

**os**: Sistema operativo con el que cuenta la laptop



In [9]:
#Tipos de datos
df.dtypes

index              int64
Company           object
TypeName          object
Ram                int64
Weight           float64
Price            float64
Touchscreen        int64
Ips                int64
ppi              float64
Cpu brand         object
HDD                int64
SSD                int64
Hybrid             int64
Flash_Storage      int64
Gpu brand         object
os                object
dtype: object

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1268 entries, 0 to 1267
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   index          1268 non-null   int64  
 1   Company        1268 non-null   object 
 2   TypeName       1268 non-null   object 
 3   Ram            1268 non-null   int64  
 4   Weight         1268 non-null   float64
 5   Price          1268 non-null   float64
 6   Touchscreen    1268 non-null   int64  
 7   Ips            1268 non-null   int64  
 8   ppi            1268 non-null   float64
 9   Cpu brand      1268 non-null   object 
 10  HDD            1268 non-null   int64  
 11  SSD            1268 non-null   int64  
 12  Hybrid         1268 non-null   int64  
 13  Flash_Storage  1268 non-null   int64  
 14  Gpu brand      1268 non-null   object 
 15  os             1268 non-null   object 
dtypes: float64(3), int64(8), object(5)
memory usage: 158.6+ KB


El dataframe no cuenta con datos nulos en ninguna de sus columnas

# Filtrado de datos relevantes

In [20]:
# Identificaion de tipos de vairables

#variables cuantitativas
df_Vcuantitativas = df[["Ram", "Weight", "Price", "Touchscreen", "Ips", "ppi", "HDD", "SSD", "Flash_Storage"]]

#Variables cualitativas
df_Vcualitativas =df[["Company", "TypeName", "Cpu brand", "Gpu brand", "os"]]

### Variables cuantitativas
- Ram (GB)
- Weight (Kg)
- Price (USD)
- TouchScreen (Binary 1/0)
- Ips (Binary 1/0)
- ppi (PPI; Pixel por pulgada)
- HDD (GB)
- SSD (GB)
- Flash Storage (GB)

In [22]:
#variables cuantitativas
df_Vcuantitativas

Ram  Weight        Price  Touchscreen  Ips         ppi   HDD  SSD  \
0       8    1.37   842.446908            0    1  226.983005     0  128   
1       8    1.34   565.286913            0    0  127.677940     0    0   
2       8    1.86   361.581390            0    0  141.211998     0  256   
3      16    1.83  1595.642953            0    1  220.534624     0  512   
4       8    1.37  1134.170774            0    1  226.983005     0  256   
...   ...     ...          ...          ...  ...         ...   ...  ...   
1263    4    1.80   401.198134            1    1  157.350512     0  128   
1264   16    1.30   942.626963            1    1  276.053530     0  512   
1265    2    1.50   144.003719            0    0  111.935204     0    0   
1266    6    2.19   480.431621            0    0  100.454670  1000    0   
1267    4    2.20   232.040927            0    0  100.454670   500    0   

      Flash_Storage  
0                 0  
1               128  
2                 0  
3                 0  
4                 0  
...             ...  
1263              0  
1264              0  
1265             64  
1266              0  
1267              0  

[1268 rows x 9 columns]

In [33]:
# Estadisticas descriptivas de las Variables cuantitativas
df_Vcuantitativas.describe().sort_values("Price", ascending=False)

Ram       Weight        Price  Touchscreen          Ips  \
max      64.000000    11.100000  3835.278083     1.000000     1.000000   
count  1268.000000  1268.000000  1268.000000  1268.000000  1268.000000   
75%       8.000000     2.320000   936.491084     0.000000     1.000000   
mean      8.461356     2.080047   707.264590     0.145110     0.282334   
50%       8.000000     2.040000   615.002782     0.000000     0.000000   
std       5.569898     0.806482   440.709489     0.352351     0.450313   
25%       4.000000     1.500000   376.673483     0.000000     0.000000   
min       1.000000     0.690000   109.417673     0.000000     0.000000   

               ppi          HDD          SSD  Flash_Storage  
max     352.465147  2000.000000  1024.000000     512.000000  
count  1268.000000  1268.000000  1268.000000    1268.000000  
75%     157.350512  1000.000000   256.000000       0.000000  
mean    145.935819   415.741325   183.634069       4.580442  
50%     141.211998     0.000000   256.000000       0.000000  
std      43.445969   517.152677   186.641125      30.615945  
25%     127.335675     0.000000     0.000000       0.000000  
min      44.019462     0.000000     0.000000       0.000000

### Variables Cualitativas
- Company
- TypeName
- Cpu Brand
- Gpu Brand
- os

In [23]:
#Variables cualitativas
df_Vcualitativas

Company            TypeName              Cpu brand Gpu brand  \
0      Apple           Ultrabook          Intel Core i5     Intel   
1      Apple           Ultrabook          Intel Core i5     Intel   
2         HP            Notebook          Intel Core i5     Intel   
3      Apple           Ultrabook          Intel Core i7       AMD   
4      Apple           Ultrabook          Intel Core i5     Intel   
...      ...                 ...                    ...       ...   
1263  Lenovo  2 in 1 Convertible          Intel Core i7     Intel   
1264  Lenovo  2 in 1 Convertible          Intel Core i7     Intel   
1265  Lenovo            Notebook  Other Intel Processor     Intel   
1266      HP            Notebook          Intel Core i7       AMD   
1267    Asus            Notebook  Other Intel Processor     Intel   

                      os  
0                    Mac  
1                    Mac  
2     Others/No OS/Linux  
3                    Mac  
4                    Mac  
...                  ...  
1263             Windows  
1264             Windows  
1265             Windows  
1266             Windows  
1267             Windows  

[1268 rows x 5 columns]

In [12]:
#Variables con respecto a la capidad de procesamiento
df_sofware = df[["Company", "TypeName", "Ram", "SSD", "HDD", "Gpu brand", "Cpu brand", "Price"]].sort_values("Price", ascending=False)
df_sofware

Company     TypeName  Ram   SSD  HDD Gpu brand              Cpu brand  \
189    Razer       Gaming   32  1000    0    Nvidia          Intel Core i7   
804    Razer       Gaming   32   512    0    Nvidia          Intel Core i7   
594   Lenovo     Notebook   32  1000    0    Nvidia  Other Intel Processor   
726       HP  Workstation   16   256    0    Nvidia  Other Intel Processor   
1034    Asus       Gaming   64  1000    0    Nvidia          Intel Core i7   
...      ...          ...  ...   ...  ...       ...                    ...   
765     Vero     Notebook    4     0    0     Intel  Other Intel Processor   
30      Asus     Notebook    2     0    0       AMD          AMD Processor   
280     Acer     Notebook    2    16    0     Intel  Other Intel Processor   
1088    Vero     Notebook    4     0    0     Intel  Other Intel Processor   
1181    Acer      Netbook    2    32    0     Intel  Other Intel Processor   

            Price  
189   3835.278083  
804   3457.975763  
594   3080.673443  
726   2759.966471  
1034  2499.627870  
...           ...  
765    127.591068  
30     125.138603  
280    125.138603  
1088   123.252091  
1181   109.417673  

[1268 rows x 8 columns]

In [ ]:
#Caracteristicas avanzadas 
df_caracteristicas_avanzadas = df[["Ips", "ppi", ]]

# Analisis del Contexto Comercial

## Compañias con los precios mas altos
- Razer
- Lenovo
- HP

## COmpañias con los precios mas bajos
- Acer
- vero 
- Asus


In [21]:
#Precios mas altos
df_precios_altos = df[["Company", "TypeName", "Price", "Ram", "SSD", "Cpu brand", "Ips"]].sort_values("Price", ascending=False)
df_precios_altos.sort_values("Price", ascending=False)

Company     TypeName        Price  Ram   SSD              Cpu brand  Ips
189    Razer       Gaming  3835.278083   32  1000          Intel Core i7    0
804    Razer       Gaming  3457.975763   32   512          Intel Core i7    0
594   Lenovo     Notebook  3080.673443   32  1000  Other Intel Processor    1
726       HP  Workstation  2759.966471   16   256  Other Intel Processor    1
1034    Asus       Gaming  2499.627870   64  1000          Intel Core i7    1
...      ...          ...          ...  ...   ...                    ...  ...
765     Vero     Notebook   127.591068    4     0  Other Intel Processor    0
280     Acer     Notebook   125.138603    2    16  Other Intel Processor    0
30      Asus     Notebook   125.138603    2     0          AMD Processor    0
1088    Vero     Notebook   123.252091    4     0  Other Intel Processor    0
1181    Acer      Netbook   109.417673    2    32  Other Intel Processor    0

[1268 rows x 7 columns]

In [32]:
#Precios mas bajos
df_precios_bajos = df[["Company", "TypeName", "Price"]]
df_precios_bajos.sort_values("Price", ascending=True)

Company     TypeName        Price
1181    Acer      Netbook   109.417673
1088    Vero     Notebook   123.252091
30      Asus     Notebook   125.138603
280     Acer     Notebook   125.138603
765     Vero     Notebook   127.591068
...      ...          ...          ...
1034    Asus       Gaming  2499.627870
726       HP  Workstation  2759.966471
594   Lenovo     Notebook  3080.673443
804    Razer       Gaming  3457.975763
189    Razer       Gaming  3835.278083

[1268 rows x 3 columns]

In [13]:
df_sofware

Company     TypeName  Ram   SSD  HDD Gpu brand              Cpu brand  \
189    Razer       Gaming   32  1000    0    Nvidia          Intel Core i7   
804    Razer       Gaming   32   512    0    Nvidia          Intel Core i7   
594   Lenovo     Notebook   32  1000    0    Nvidia  Other Intel Processor   
726       HP  Workstation   16   256    0    Nvidia  Other Intel Processor   
1034    Asus       Gaming   64  1000    0    Nvidia          Intel Core i7   
...      ...          ...  ...   ...  ...       ...                    ...   
765     Vero     Notebook    4     0    0     Intel  Other Intel Processor   
30      Asus     Notebook    2     0    0       AMD          AMD Processor   
280     Acer     Notebook    2    16    0     Intel  Other Intel Processor   
1088    Vero     Notebook    4     0    0     Intel  Other Intel Processor   
1181    Acer      Netbook    2    32    0     Intel  Other Intel Processor   

            Price  
189   3835.278083  
804   3457.975763  
594   3080.673443  
726   2759.966471  
1034  2499.627870  
...           ...  
765    127.591068  
30     125.138603  
280    125.138603  
1088   123.252091  
1181   109.417673  

[1268 rows x 8 columns]

In [22]:
#Compañias con las laptos de mas alta calidad en relacion con si capacidad/Procesamiento - precio
df_compañias_highquality = df[df["Company"].isin(["Razer", "Lenovo", "HP"])].sort_values("Price", ascending=False)
df_compañias_highquality


Company     TypeName  Ram  Weight        Price  Touchscreen  Ips  \
189    Razer       Gaming   32    3.49  3835.278083            1    0   
804    Razer       Gaming   32    3.49  3457.975763            1    0   
594   Lenovo     Notebook   32    2.50  3080.673443            0    1   
726       HP  Workstation   16    3.00  2759.966471            0    1   
1104      HP  Workstation    8    3.00  2483.529638            0    1   
...      ...          ...  ...     ...          ...          ...  ...   
1265  Lenovo     Notebook    2    1.50   144.003719            0    0   
1251  Lenovo     Notebook    2    1.50   144.003719            0    0   
1233      HP      Netbook    2    1.17   131.426975            0    0   
1261      HP      Netbook    2    1.17   131.426975            0    0   
1247      HP      Netbook    2    1.17   131.426975            0    0   

             ppi              Cpu brand  HDD   SSD  Flash_Storage Gpu brand  \
189   254.671349          Intel Core i7    0  1000              0    Nvidia   
804   254.671349          Intel Core i7    0   512              0    Nvidia   
594   282.423996  Other Intel Processor    0  1000              0    Nvidia   
726   127.335675  Other Intel Processor    0   256              0    Nvidia   
1104  127.335675          Intel Core i7    0   256              0    Nvidia   
...          ...                    ...  ...   ...            ...       ...   
1265  111.935204  Other Intel Processor    0     0             64     Intel   
1251  111.935204  Other Intel Processor    0     0             64     Intel   
1233  135.094211  Other Intel Processor    0     0             32     Intel   
1261  135.094211  Other Intel Processor    0     0             32     Intel   
1247  135.094211  Other Intel Processor    0     0             32     Intel   

           os  
189   Windows  
804   Windows  
594   Windows  
726   Windows  
1104  Windows  
...       ...  
1265  Windows  
1251  Windows  
1233  Windows  
1261  Windows  
1247  Windows  

[563 rows x 14 columns]

In [ ]:
# Compañias de las laptops de mas bajo precio en relacion con su capacidad/Procesamiento - precio


In [ ]:
df_compañias_lowquality = df[df["Company"].isin(["Razer", "Lenovo", "HP"])].sort_values("Price", ascending=False)

# Conclusiones
## Contexto comercial
  ### Variables
  - Ram
  - SSD
  - 